BlueprintUI/Sources/Layout/Builder/GridRowChild.swift

```diff
@@ -0,0 +1,58 @@
+import UIKit
+
+// MARK: - child modeling -
+extension GridRow {
+    /// A child of a `GridRow`.
+    public struct Child {
+        // MARK: - properties -
+        /// The element displayed in the `Grid`.
+        public var element: Element
+        /// A unique identifier for the child.
+        public var key: AnyHashable?
+        // The sizing for the element.
+        public var width: Width
+
+        // MARK: - initialialization -
+        public init(width: Width, key: AnyHashable? = nil, element: Element) {
+            self.element = element
+            self.key = key
+            self.width = width
+        }
+    }
+
+    /// The sizing and content of a `GridRow` child.
+    public enum Width: Equatable {
+        /// Assign the child a fixed width equal to the payload.
+        case absolute(CGFloat)
+        /// Assign the child a proportional width of the available layout width. Note that proportional children
+        /// take proportional shares of the available layout width.
+        ///
+        /// ## Example:
+        ///     Available layout width: 100
+        ///     Child A: .proportional(1)  -> 25 (100 * 1/4)
+        ///     Child B: .proportional(3) -> 75 (100 * 3/4)
+        ///
+        /// ## Example:
+        ///     Available layout width: 100
+        ///     Child A: .proportional(0.25)  -> 25 (100 * 1/4)
+        ///     Child B: .proportional(0.75) -> 75 (100 * 3/4)
+        case proportional(CGFloat)
+    }
+}
+
+extension Element {
+    /// Wraps an element with a `GridRowChild` in order to provide meta information that a `GridRow` can aply to its layout.
+    /// - Parameters:
+    ///   - key: A unique identifier for the child.
+    ///   - width: The sizing for the element.
+    /// - Returns: `GridRowChild`
+    public func gridRowChild(key: AnyHashable? = nil, width: GridRow.Width) -> GridRow.Child {
```

Comment:
I'm curious for others' thoughts here, but I wonder if we want to _not_ add these kinds of helpers; since they're only valid in a given context (you using the result in a grid row (etc). If you use the wrong helper, or an element uses the helper is presented in a different context; it basically is a silent failure. 

Eg, maybe we just want to force people to explicitly return a `GridRow.Child`; but then again, you can still fuck that up in the same exact ways, since everything is type erased anyways... So maybe this is fine! Thoughts?

Reply:
So because we changed `ElementBuilder` to use a corresponding `Child` (i.e. GridRow.Child) as its currency and that `Child` no longer conforms to `Element`, that means this modifier can really only be used inside of its supported container. If you use it in a mismatching container, or an API that takes in an element, you will get a compile time error.

Reply:
I think what we have here is worth the trade offs (e.g. less code/nesting for common use case), and as @ldstreet pointed out, you can't really use it in the wrong context since it's not an `Element`.

Reply:
Ah cool! Sounds good.